corehq/apps/accounting/tests/test_invoicing.py

```diff
@@ -450,6 +454,107 @@ def test_community_over_limit(self):
         self.assertEqual(user_line_item.total, num_to_charge * self.user_rate.per_excess_fee)
 
 
+class TestFormSubmittingMobileWorkerLineItem(BaseInvoiceTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        feature_rates = generator.default_feature_rates() + [
+            FeatureRate.objects.create(
+                feature=Feature.objects.get(name=FeatureType.FORM_SUBMITTING_MOBILE_WORKER),
+                monthly_limit=10,
+                per_excess_fee=Decimal('3.00')
+            )
+        ]
+        plan_version = generator.custom_plan_version(feature_rates=feature_rates)
+        cls.subscription = generator.generate_domain_subscription(
+            cls.account,
+            cls.domain,
+            date_start=cls.subscription_start_date,
+            date_end=cls.subscription_end_date,
+            is_active=cls.subscription_is_active,
+            plan_version=plan_version
+        )
+
+    def setUp(self):
+        super(TestFormSubmittingMobileWorkerLineItem, self).setUp()
+        self.worker_rate = self.subscription.plan_version.feature_rates.filter(
+            feature__feature_type=FeatureType.FORM_SUBMITTING_MOBILE_WORKER)[:1].get()
+
+    def create_worker_history(self, history_cls, record_date, num_workers):
+        history_cls.objects.create(
+            domain=self.domain,
+            record_date=record_date,
+            num_users=num_workers
+        )
+
+    def test_under_limit(self):
```

Comment:
That's a really good point. I was following what had been done for testing `WebUserLineItemFactory`, but that at least has a chunk of different logic, where `FormSubmittingMobileWorkerLineItemFactory` really is just a couple lines of difference.

FWIW we _don't_ currently have `TestLineItemFactory` but we could. I would see that being where testing for `under_limit` and `over_limit` happens. The one other thing that would want to be tested, which currently gets looked at implicitly, is whether adding the right `Feature` on the `SoftwarePlanVersion` causes that feature to show up correctly on an invoice. I think those might make sense as asserts on `TestLineItemFactory`, something like `test_form_submitting_mobile_worker_feature_on_invoice`.

Reply:
Rather than rework a bulk of the invoicing test suite, I opted for what I felt like was a happy-medium. I restructured so what we're testing are mainly the things that are based on the Form-Submitting Mobile Worker `FeatureRate`, rather than every generic line item property, and then the `unit_description`. https://github.com/dimagi/commcare-hq/pull/34868/commits/0afcabb5d356eea7eb341995e07682aa97dd8d66

Reply:
I do notice we still preserved the tests for over and under limits, while it sounds like we're mainly just trying to verify that the specific changes for form submitting workers is different. Do you think the benefit of maintaining these tests outweighs the duplication and complexity of tests that are, at least by their name, trying to verify other things? Would it be possible to replace these with a single test verifying the unique features instead?